launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FileUploadServlet.java

```diff
@@ -0,0 +1,294 @@
+    userFeedEntity.setProperty(LANGUAGE, language);
+    userFeedEntity.setProperty(EMAIL, email);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key entityKey = datastore.put(userFeedEntity);
+
+    String entityId = KeyFactory.keyToString(entityKey);
+    String mp3Link = makeMp3Link(entityId);
+
+    // Create embedded entity to store MP3 data
+    EmbeddedEntity mp3 = new EmbeddedEntity();
+    mp3.setProperty(ID, entityId);
+    mp3.setProperty(MP3_LINK, mp3Link);
+    mp3.setProperty(EMAIL, email);
+
+    // Retrieve entity from Datastore
+    Entity savedEntity;
+    try {
+      savedEntity = datastore.get(entityKey);
+    } catch(EntityNotFoundException e) {
+      throw new IOException("Entity not found in Datastore.");
+    }
+
+    // Generate xml string
+    // Note: this will be modified after merging because we will be modifying the existing XML for a feed for the MP3 upload feature
+    RSS rssFeed = new RSS(podcastTitle, description, language, email, mp3Link);
+    String xmlString;
+    try {
+      xmlString = RSS.toXmlString(rssFeed);
+      savedEntity.setProperty(XML_STRING ,xmlString);
+    } catch(IOException e){
+      throw new IOException("Unable to create XML string.");
+    }
+
+    // Update entity by adding embedded mp3 entity as a property
+    savedEntity.setProperty(MP3, mp3);
+    datastore.put(savedEntity);
+
+    //write the file upload form
+    String formHtml = generateSignedPostPolicyV4(PROJECT_ID, BUCKET_NAME, entityId);
+    res.setContentType("text/html");
+    res.getWriter().println(formHtml);
+  }
+
+  /**
+   * Generate policy for directly uploading a file to Cloud Storage via HTML form.
+   * @return HTML form (as a String) for uploading MP3
+   */
+  public String generateSignedPostPolicyV4 (String projectId, String bucketName, String blobName) throws IOException {
+    // String blobName = entity ID from Datastore, as the name of object uploaded to GCS
+
+    String errorMessage = "Policy is null.";
+    Storage storage;
+    PostPolicyV4.PostFieldsV4 fields;
+    PostPolicyV4 policy = null;
+    String myRedirectUrl = "";
+
+    try {
+      // Prepare credentials and API key
+      String keyFileName = "launchpod-step18-2020-47434aafba88.json";
+      ClassLoader classLoader = getClass().getClassLoader();
+      File file = new File(classLoader.getResource(keyFileName).getFile());
+      GoogleCredentials credentials = GoogleCredentials.fromStream(new FileInputStream(file));
+
+      storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+
+      myRedirectUrl = GENERATE_RSS_URL + blobName;
+      fields =
+          PostPolicyV4.PostFieldsV4.newBuilder().setSuccessActionRedirect(myRedirectUrl).build();
+
+      policy =
+        storage.generateSignedPostPolicyV4(
+            BlobInfo.newBuilder(bucketName, blobName).build(), 10, TimeUnit.MINUTES, fields);
+    } catch(Exception e) {
```

Comment:
let's specify what exception we're catching. Exception is too general and will catch all exceptions, even if you didn't intend to catch them